src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -578,19 +578,32 @@ def update_iot_hub_custom(instance,
     if fileupload_notification_ttl is not None:
         ttl = timedelta(hours=fileupload_notification_ttl)
         instance.properties.messaging_endpoints['fileNotifications'].ttl_as_iso8601 = ttl
-    # if setting a fileupload storage identity or changing fileupload to identity-based
-    if fileupload_storage_identity or fileupload_storage_authentication_type == AuthenticationType.IdentityBased.value:
-        _validate_fileupload_identity(instance, fileupload_storage_identity)
-
-    default_storage_endpoint = _process_fileupload_args(
-        instance.properties.storage_endpoints['$default'],
-        fileupload_storage_connectionstring,
-        fileupload_storage_container_name,
-        fileupload_sas_ttl,
-        fileupload_storage_authentication_type,
-        fileupload_storage_container_uri,
-        fileupload_storage_identity,
-    )
+    # only bother with $default storage endpoint checking if modifying fileupload params
+    if any([
+            fileupload_storage_connectionstring, fileupload_storage_container_name, fileupload_sas_ttl,
+            fileupload_storage_authentication_type, fileupload_storage_container_uri, fileupload_storage_identity]):
+        default_storage_endpoint = instance.properties.storage_endpoints.get('$default', None)
+        # no default storage endpoint, either recreate with existing params or throw an error
+        if not default_storage_endpoint:
+            if not all([fileupload_storage_connectionstring, fileupload_storage_container_name]):
+                raise AzCLIError('This hub has no default storage endpoint for file upload.\n'
+                                 'Please recreate your default storage endpoint by running '
+                                 '`az iot hub update --name {hub_name} --fcs {storage_connection_string} --fc {storage_container_name}`')
```

Comment:
Could we use the `UserFault` instead of `AzCLIError` ?

Reply:
Sure - this one's a bit tough because I'm not sure what the correct error class would be - technically it is a user fault for trying to modify a storage endpoint that doesn't exist, but in the context of the command, the fault would be an `RequiredArgumentMissingError`.

For the time being we can classify it as an `UnclassifiedUserFault` since it's more complicated than just incorrect arguments or a misconfigured resource.